mmedit/datasets/pipelines/augmentation.py

```diff
@@ -965,15 +963,18 @@ def __call__(self, results):
         # add the corresponding file paths
         lq_path_root = results['lq_path']
         gt_path_root = results['gt_path']
+
+        # retrieve the file names in the directory
+        filenames = sorted(
```

Comment:
* scandir should only do once. Otherwise disk access will become speed bottleneck
* This will fail if the backend is ceph.

So I changed my mind: support a start_index, instead of scanning folders

Reply:
Okay.